### ISSUE TITLE
Seing shadows of invised spys on the floor above you.

### ISSUE THREAD

**Issue by WiFiDi** [fortressforever/fortressforever#200]
I've seen this a few times recently. most of the time i just ignore it im pretty sure this problem has been around awhile and im not sure if its source 2006 or its somthing that can be fixed. so i post here just incase.

if you guys cant figure out what im talking about im gonna see if i can find a screenshot.

**Comment by WiFiDi** [comment]
DOH forget what i said i realized it most likely has to do with medieval b13 not the game... (facepalm)

**Comment by squeek502** [comment]
No, it's a legitimate bug that plagues the Source engine (not sure about newer versions). We tried to minimize it by locking the shadow angle at 90 degrees (straight down), but it still pops up in certain places. For example, I know that if you crouch underneath the middle ramp in the shutdown2 flagroom, your  shadow will appear on top of the ramp.

**Comment by hlstriker** [comment]
Some reason I remember "fixing" this in CS:S by tweaking maximum shadow cast distance (as well as setting the angle 90 degrees like you said). Not sure if it can be tweaked further in FF using the cast distance.

**Comment by bmk10** [comment]
ComputeFx in 2007/2013 source versions of c_sdk_player or C_SDK_Player or it might be C_FF_Player in ff case, should ComputeFxBlend is the shadow cast function looks like something like (for tf)  
void C_SDKPlayer::ComputeFxBlend( void )
{
    BaseClass::ComputeFxBlend();

```
if ( GetPlayerClass()->IsClass( TF_CLASS_SPY ) )
{
    float flInvisible = GetPercentInvisible();
    if ( flInvisible != 0.0f )
    {
        // Tell our shadow
        ClientShadowHandle_t hShadow = GetShadowHandle();
        if ( hShadow != CLIENTSHADOW_INVALID_HANDLE )
        {
            g_pClientShadowMgr->SetFalloffBias( hShadow, flInvisible * 255 );
        }
    }
}
```

}

**Comment by WiFiDi** [comment]
yea, not totally off the wall bonkers this is good news i think. :stuck_out_tongue_winking_eye:
